include/tester/vtcp/VTCPRequestHandler.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include<functional>
+#include<memory>
+#include<unordered_map>
+
+#include"generic/RequestHandler.hpp"
+#include"VTCPOpcode.hpp"
+
+class VTCPRequestHandler : public RequestHandler
+{
+public:
+    VTCPRequestHandler():
+        RequestHandler()
+    {
+        _handlers.emplace(VTCPOpcode::VTCP_CONNECT_REQ,std::bind(&VTCPRequestHandler::onVtcpConnect, this, std::placeholders::_1, std::placeholders::_2));
+        _handlers.emplace(VTCPOpcode::VTCP_DISCONNECT_REQ,std::bind(&VTCPRequestHandler::onVtcpDisconnect, this, std::placeholders::_1, std::placeholders::_2));
+        _handlers.emplace(VTCPOpcode::VTCP_MANUAL_TEST_REQ,std::bind(&VTCPRequestHandler::onVtcpManualTest, this, std::placeholders::_1, std::placeholders::_2));
+        _handlers.emplace(VTCPOpcode::VTCP_AUTO_TEST_REQ,std::bind(&VTCPRequestHandler::onVtcpAutoTest, this, std::placeholders::_1, std::placeholders::_2));
+    }
+
+    void handle(std::shared_ptr<TCPSession> sender, const Message& request) override
+    {
+        try
+        {
+            auto opcode = static_cast<VTCPOpcode>(request.readInteger());
+            _handlers.at(opcode)(std::move(sender), request);
+        }
+        catch(const std::exception& e)
+        {
+            std::cout << e.what() << std::endl;
+        }
+    }
+
+    ~VTCPRequestHandler() = default;
+
+    void onVtcpConnect(std::shared_ptr<TCPSession> sender, const Message & data)
+    {
+        std::cout << "New client connected" << std::endl;
+        Message result;
+        result.push(static_cast<int>(VTCPOpcode::VTCP_CONNECT_RES));
+        std::string result_msg = "Hello new client";
+        result.push(result_msg);
+        result.pushSize();
+        sender->send(result.getPayload());
+    }
+
+    void onVtcpDisconnect(std::shared_ptr<TCPSession> sender, const Message & data)
+    {
+        std::cout << "Client disconnected" << std::endl;
+
+        //sender->onDisconnect();
+    }
+
+    void onVtcpAutoTest(std::shared_ptr<TCPSession> sender, const Message & request)
```

Comment:
Move method below `onVtcpManualTest`

Reply:
I change the order of the `_handlers.emplace(...)` in the ctor that `onVtcpManualTest` will be after `onVtcpAutoTest`

Reply:
ok